pymodbus/server/async_io.py

```diff
@@ -223,8 +223,8 @@ def __init__(
         self.handle_local_echo = False
         if isinstance(identity, ModbusDeviceIdentification):
             self.control.Identity.update(identity)
-
-        self.framer = FRAMER_NAME_TO_CLASS[framer]
+        # Support mapping of FramerType to a Framer class, or a Framer class
```

Comment:
This is only the server, if we are to include this feature again, at least it must also support client and sync client.

Reply:
OK, I think I've generalized this to all servers and clients.  I couldn't figure out where in the tests to be able to add an "evil" framer, though...

Reply:
Did you look at the test directory, it has a sub directory called framer, which is where framer testing goes. Please do not change existing tests, but add tests.

FramerBase is an internal class which we do not want public, because having it public makes it a lot harder to change, please see my bullet list.

Reply:
I did look at it, but didn't find any tests of framers in a full I/O cycle.  I'll look again.

Agreed; I don't think FramerBase should be public -- users can derive "evil" framers from any of the existing public FramerXxx classes.